<?xml version="1.0"?>
<launch>

    <arg name="robot_name" default=""/>

    <!-- pops out a gui from which you can send position ctrl commands to a robot -->
    <group ns="$(arg robot_name)">
        <!-- send fake joint values from gui -->
        <node pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" name="position_ctrl_cmd_gui">
            <remap from="joint_states" to="fake_joint_states"/>
        </node>

        <!-- republish the fake joint states to floats to send commands to the arm joints in position ctrl -->
        <node pkg="pybullet_ros" type="joint_states_to_float" name="position_ctrl_repub" output="screen">
            <param name="robot_name" value="$(arg robot_name)"/>
        </node>
    </group>

</launch>
